controller/controlplane/controller_watch.go

```diff
@@ -74,18 +75,17 @@ func (r *Reconciler) validatingWebhookConfigurationHasControlPlaneOwner(obj clie
 // ClusterScopedObjHasControlPlaneOwner checks if the cluster-scoped object has a control plane owner.
 // The check is performed through the managed-by-name label.
 func (r *Reconciler) ClusterScopedObjHasControlPlaneOwner(ctx context.Context, obj client.Object) bool {
-	controlplaneList := &operatorv1beta1.ControlPlaneList{}
-	if err := r.Client.List(ctx, controlplaneList); err != nil {
+	var controlplaneList operatorv1beta1.ControlPlaneList
```

Comment:
What's the point of this change?

Reply:
No need to allocate here. We can use a var on stack. (note: this is probably going to be optimized in the compiler during the escape analysis: https://tip.golang.org/src/cmd/compile/internal/escape/escape.go, but if we can let's mark this as stack allocated anyway).

WDYT?

Reply:
ok, it makes sense 👍